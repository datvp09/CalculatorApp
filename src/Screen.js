import React from "react"
import { View, Text } from "react-native"
import { numberButtons, mathButtons, isIpX } from "../src/constants"
import styles from "../src/styles"
import Button from "../src/Button"
import { connect } from "react-redux"
import { _handleEvent } from "./reducer"

const Screen = props => (
  <View style={[styles.container, isIpX && { marginBottom: 34 }]}>
    <View style={styles.statusBar} />
    <View style={styles.placeHolderOutput}>
      <Text style={styles.txtDefault}>{props.output}</Text>
    </View>
    <View style={styles.contRow}>
      <View style={styles.flexNumbers}>
        {numberButtons.map((row, index) => (
          <View key={index} style={styles.contRow}>
            {row.map((col, index) => (
              <Button
                key={index}
                onPress={() => _handleEvent(col, props)}
                text={col}
              />
            ))}
          </View>
        ))}
      </View>
      <View style={styles.container}>
        {mathButtons.map((row, index) => (
          <Button
            style={{ marginRight: 10 }}
            key={index}
            onPress={() => _handleEvent(row, props)}
            text={row}
          />
        ))}
      </View>
    </View>
  </View>
)

const mapStateToProps = state => ({ output: state.calReducer.output })
const mapDispatchToProps = dispatch => {
  const {
    clear,
    evaluate,
    replace,
    concat,
    concatwithout
  } = require("./actions")
  return {
    clear: () => dispatch(clear()),
    evaluate: () => dispatch(evaluate()),
    replace: value => dispatch(replace(value)),
    concat: value => dispatch(concat(value)),
    concatwithout: value => dispatch(concatwithout(value))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Screen)
