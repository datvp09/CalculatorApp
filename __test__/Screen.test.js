import "react-native"
import React from "react"
import Screen from "../src/Screen"
import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import configureMockStore from "redux-mock-store"

const mockStore = configureMockStore()
const store = mockStore({ calReducer: { output: "0" } })

describe("Screen", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const component = renderer
        .create(
          <Provider store={store}>
            <Screen />
          </Provider>
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
