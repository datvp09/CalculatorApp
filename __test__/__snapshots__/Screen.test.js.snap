// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Rendering should match to snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      false,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 60,
        }
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              7
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              8
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              9
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              5
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              6
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              1
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              2
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              3
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              C
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              0
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 5,
                "flex": 1,
                "marginBottom": 10,
                "marginLeft": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ddd",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 5,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 26,
                }
              }
            >
              .
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            /
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            *
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            +
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            -
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "borderRadius": 5,
              "flex": 1,
              "marginBottom": 10,
              "marginLeft": 10,
            },
            Object {
              "marginRight": 10,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#ddd",
                "flex": 1,
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 26,
              }
            }
          >
            =
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
